services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeServiceITests.kt

```diff
@@ -907,6 +913,21 @@ class HankeServiceITests : DatabaseTest() {
         assertThat(alue.geometriat).isNotNull
     }
 
+    @Test
+    fun `generateHankeWithApplication generates hanke based on application`() {
+        val inputApplication = AlluDataFactory.createApplication()
+        every { applicationService.create(any(), USER_NAME) } returnsArgument 0
+
+        val result = hankeService.generateHankeWithApplication(inputApplication, USER_NAME)
+
+        with(result) {
+            val application = applications.first()
+            assertThat(hanke.hankeTunnus).isEqualTo(application.hankeTunnus)
+            assertThat(hanke.nimi).isEqualTo(application.applicationData.name)
```

Comment:
So this checks that the names in the created application and the generated hanke are the same. Should this also check that it's the name from the input application?